home-work-objects-in-details.js

```diff
@@ -0,0 +1,72 @@
+// 1. Задача про обчислення різниці часу
+
+function durationBetweenDates(firstDate = '24 Aug 1991', secondDate = '24 Aug 1991', dimension = 'days') {
+  const startDate = new Date(firstDate);
+  const endDate = new Date(secondDate);
+
+  const timeDiff = Math.abs(endDate - startDate); // time difference in milliseconds
+
+  const dimensionMap = {
+    days: 1000 * 60 * 60 * 24,
+    hours: 1000 * 60 * 60,
+    minutes: 1000 * 60,
+    seconds: 1000
+  };
+
+  const dimensionDiff = Math.floor(timeDiff / dimensionMap[dimension]);
```

Comment:
We will have a strange error when someone uses this function with an unsupported dimension. Let's fix it.

Reply:
On 16th line added check for incorrect dimension